scene/gui/color_picker.h

```diff
@@ -75,6 +75,24 @@ class ColorPresetButton : public BaseButton {
 	~ColorPresetButton();
 };
 
+class SwatchColorButton : public ColorPresetButton {
+	GDCLASS(SwatchColorButton, ColorPresetButton);
+
+public:
+	virtual String get_tooltip(const Point2 &p_pos) const override;
+
+	using ColorPresetButton::ColorPresetButton;
+};
+
+class RecentColorButton : public ColorPresetButton {
+	GDCLASS(RecentColorButton, ColorPresetButton);
+
+public:
+	virtual String get_tooltip(const Point2 &p_pos) const override;
+
+	using ColorPresetButton::ColorPresetButton;
+};
```

Comment:
You could add a `bool recent` to ColorPresetButton; this way you won't need new classes.